### PR TITLE
React-tree/bugfix--aside-and-actions-as-slot

### PR DESCRIPTION
**Pull request by bsunderhus**
## Previous Behavior

A `TreeItemAside` component was created on the intent of giving more power to the user to decide on the visibility of actions/aside slots by providing a `visible` property on the `TreeItemAside` component.

Together with that, both `aside` and `actions` slots were removed from the layout components being the user's responsibility to properly position `TreeItemAside`

## New Behavior

1. rollback from using a slot for `aside` and `actions` as those elements should have a proper internal position that has to be ensure by the layout component
2. adds a `visible` property in the `actions` slot to ensure that the user can control actions visibility just like `TreeItemAside` allowed
3. removes `TreeItemAside` component as it's not required if we have slots on the layout components
4. by ensuring positioning of the slots not visual regressions are introduced and we're once again aligned with design spec with a smaller codebase

## Related Issue(s)

Fixes some visual regressions pointed out by our design team where the actions and aside elements are not properly visually positioned.

### CODE REVIEW

File: packages/react-components/react-tree/src/components/TreeItem/useTreeItemStyles.styles.ts [hunk 1]
```diff
@@ -22,14 +22,9 @@ const useRootStyles = makeStyles({
   base: {
     position: 'relative',
     cursor: 'pointer',
-    display: 'grid',
+    display: 'flex',
```
Comment (bsunderhus) [inline packages/react-components/react-tree/src/components/TreeItem/useTreeItemStyles.styles.ts:25]: `TreeItem` becomes a column oriented flexbox. As all possible visual items are now part of the layout component the `TreeItem` itself ends up having only 2 possible children:

1. Layout component
2. Subtree

File: packages/react-components/react-tree/src/components/TreeItemLayout/TreeItemLayout.types.ts [hunk 2]
```diff
@@ -15,6 +11,30 @@ export type TreeItemLayoutSlots = {
...
+         * Forces visibility of the aside/action content
+         */
+        visible?: boolean;
+      }
+    >
+  >;
+};
+
+export type TreeItemLayoutInternalSlots = TreeItemLayoutSlots & {
+  actions?: Slot<'div'>;
```
Comment (bsunderhus) [inline packages/react-components/react-tree/src/components/TreeItemLayout/TreeItemLayout.types.ts:37]: This is required to ensure we're properly removing the `visible` property, that is available by the external props, from the actual internal slot (that should not have that property propagated to the DOM)

File: packages/react-components/react-tree/src/components/TreeItemPersonaLayout/TreeItemPersonaLayout.types.ts [hunk 3]
```diff
@@ -24,10 +21,10 @@ export type TreeItemPersonaLayoutSlots = {
    * Secondary text that describes or complements the main text
    */
   description?: Slot<'div'>;
-  /**
-   * A layout wrapper for the main and description slots
-   */
-  content: NonNullable<Slot<'div'>>;
+};
+
+export type TreeItemPersonaLayoutInternalSlots = TreeItemPersonaLayoutSlots & {
```
Comment (bsunderhus) [inline packages/react-components/react-tree/src/components/TreeItemPersonaLayout/TreeItemPersonaLayout.types.ts:26]: This is required to ensure we're properly removing the `visible` property, that is available by the external props, from the actual internal slot (that should not have that property propagated to the DOM)

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/a5d2752d447c124abcd2cee2c36891b6c1b83a83'>a5d2752d447c124abcd2cee2c36891b6c1b83a83</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Avatar-mount_base.html">599</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Avatar-mount.html">591</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Button-mount_base.html">296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Button-mount.html">293</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Field-mount_base.html">1096</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Field-mount.html">1046</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProvider-mount_base.html">640</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProvider-mount.html">661</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProviderWithTheme-mount.html">83</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">62</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">67</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">69</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/InfoButton-mount_base.html">8</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/InfoButton-mount.html">12</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/MakeStyles-mount_base.html">849</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/MakeStyles-mount.html">854</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Persona-mount_base.html">1656</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/Persona-mount.html">1646</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/SpinButton-mount_base.html">1345</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/28318/perf-test-react-components/SpinButton-mount.html">1317</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by layershifter — APPROVED** [review]

**Pull request merged**